lib/block_view_lockups.go

```diff
@@ -818,109 +1174,541 @@ func (bav *UtxoView) _connectCoinLockup(
 		bav._setProfileEntryMappings(profileEntry)
 	}
 
-	// By now we know the transaction to be valid. We now source yield information from either
-	// the profile's yield curve or the raw DeSo yield curve. Because there's some choice in how
-	// to determine the yield when the lockup duration falls between two profile-specified yield curve
-	// points, we return here the two local points and choose/interpolate between them below.
-	leftYieldCurvePoint, rightYieldCurvePoint, err :=
-		bav.GetLocalYieldCurvePoints(profilePKID, lockupDurationNanoSeconds)
-	if err != nil {
-		return 0, 0, nil,
-			errors.Wrap(err, "_connectCoinLockup failed to fetch yield curve points")
-	}
+	// If this is an unvested lockup, compute any accrued yield.
+	// In the vested lockup case, the yield earned is always zero.
+	yieldFromTxn := uint256.NewInt()
+	if txMeta.UnlockTimestampNanoSecs == txMeta.VestingEndTimestampNanoSecs {
```

Comment:
Let's add one more safeguard here. Can you first look up if the profile has any yield curve points and add that as a check before we enter this logic? The reason is I just want profiles without yield settings to be super duper safe. So something like this:
```
if profileEnablesYield && txMeta.UnlockTimestampNanoSecs == txMeta.VestingEndTimestampNanoSecs {
```

Reply:
This will also protect DESO txns from entering this block at all, which is good.